src/main/scala/mesosphere/marathon/core/deployment/impl/StartingBehavior.scala

```diff
@@ -46,7 +46,9 @@ trait StartingBehavior extends ReadinessBehavior with StrictLogging { this: Acto
     case InstanceChanged(id, `version`, `pathId`, condition: Condition, instance) if condition.isTerminal || instance.isReservedTerminal =>
       logger.warn(s"New instance [$id] failed during app ${runSpec.id.toString} scaling, queueing another instance")
       instanceTerminated(id)
-      launchQueue.add(runSpec, 1).pipeTo(self)
+      if (instance.state.goal != Goal.Running) {
```

Comment:
to avoid overscaling

Reply:
What do you think about pulling in the logic from `LaunchQueue.add`? We are going to do that anyways and it would make the scheduling more explicit:
```scala
instance.state.goal match {
  case Goal.Stopped => setGoal(instance, Goal.Running) // aka reschedule
  case Goal.Decommisioned => instanceTracker.schedule(newInstance)
}
```

Reply:
I agree with @jeschkies in general - the less logic `LaunchQueue` has, the easier it is to remove later. However, 

> `case Goal.Stopped => setGoal(instance, Goal.Running)`

this is a strange line. Somebody (presumably user) set the new instance goal, and we simply override this?  What's the use case here?

Reply:
This is a good question. For now `LaunchQueue.add` is doing exactly that. However, do we want this?

AFAIK the only time a user can set that goal to `Goal.Stopped` is with a non-wipe kill. The user expects the task to start again. Now, does that mean the was was killed before? @timcharper suggested, yes and proposed also a "goal incarnation". I'd argue we don't have to guarantee a restart.

That said, in this case we have a terminal condition. That means we fulfilled the `Goal.Stopped`. At least I think so.

Reply:
This use case:
- user sets `Goal.Stopped`, so that the task is killed, instance kept and no replacement is started
we *don't have it in our current v2 API* right now. Maybe we'll discover a good use case for it later, then we can get back to "goal incarnation" discussion.

Reply:
> user sets Goal.Stopped, so that the task is killed, instance kept and no replacement is started

Hm, I'm not sure. Marathon < 1.8 would just scale and start a replacement. So maybe we should do just that. Start a replacement instance for ephemeral instance at least. For persistent instance Marathon would restart the instance. We could transfer the reservation. Not sure if that is the way to go. Anyways, `launchQueue.add` would retain the old behaviour, ie set the goal to `Running` again.

Reply:
> Start a replacement instance for ephemeral instance at least. For persistent instance, Marathon would restart the instance.

I respectfully disagree. For ephemeral/persistent tasks what you want (most of the time) is to kill the task (and increase the incarnation count). The only reason to get rid of the instance completely is when it should disappear aka `wipe=true` case.